src/liballoc/arc.rs

```diff
@@ -581,7 +581,7 @@ impl<T: Eq> Eq for Arc<T> {}
 
 impl<T: fmt::Show> fmt::Show for Arc<T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        (**self).fmt(f)
+        write!(f, "Arc {{ {:?} }}", (**self))
```

Comment:
They're different since Box is a tuple struct, and Arc and Rc aren't. I can make then equivalent, though.